### DISCUSSION TITLE
I tried to install this

### DISCUSSION THREAD

**Discussion by TheAmazinRaven** [jitinnair1/gradfolio#12] [General]
I would love to use this theme but every time I've tried to install it, it tells me I need a gem file or an SEO tag. I'm not sure how to fix this. Could you help me or explain?

**Comment by jitinnair1** [comment]
I'm assuming that by installation, you meant getting the theme to run locally. Ideally, if you run using `jekyll serve --incremental --trace`, it should not ask for a gem file. Could you please share a screenshot of the error message you got? Or let me know if I have misunderstood the situation.

**Reply by TheAmazinRaven** [reply]
Yes I sure will. Thank you so much for responding. I will get that to you asap.

**Reply by TheAmazinRaven** [reply]
![traceWontWork](https://user-images.githubusercontent.com/33698820/139118185-7e59bf8b-be69-4a50-a3ea-25a47ae32259.png)

I attempted this command, it says command not found.

**Reply by TheAmazinRaven** [reply]
![noSEOTag](https://user-images.githubusercontent.com/33698820/139119280-7e1f5dd3-36fb-4e17-82ed-d7d56ee123cd.png)

I see where I mistyped earlier but now it's saying no SEO tag.

**Reply by jitinnair1** [reply]
I think this is because one of the plugins is not installed (`jekyll-seo-tag` in this case). You can install it by running `gem install jekyll-seo-tag`. After installing the plugin, `jekyll serve --incremental --trace` should run without errors.

**Reply by TheAmazinRaven** [reply]
I installed the jekyll seo tag, it's still giving me the same error message. :(

**Reply by jitinnair1** [reply]
Let me know if you get errors even after installing the missing plugin. If you get error messages for any other missing plugins, you can install all dependencies at once by running `gem install jekyll-seo-tag jekyll-paginate jekyll-email-protect jekyll-target-blank`

Thanks for pointing this out. I'll add a note about dependencies in the README section of the repo.

**Reply by jitinnair1** [reply]
> I installed the jekyll seo tag, it's still giving me the same error message. :(

Please share a screenshot

**Reply by TheAmazinRaven** [reply]
I installed all of the dependencies and that did not give me any errors. Local host:4000 is still not working but when I preview the website on GH Pages, there are no stylings. It has accepted my edits (yay!) but no stylings.

**Reply by TheAmazinRaven** [reply]
> > I installed the jekyll seo tag, it's still giving me the same error message. :(
> 
> Please share a screenshot

I wrote that I got the error message right before installing all of the dependencies. I really appreciate your help w/ this.

**Reply by TheAmazinRaven** [reply]
They seo tag files are conflicting but I don't know how to edit this without a gemfile. It's not in `config.yml`

![conflctingSEO](https://user-images.githubusercontent.com/33698820/139123424-9f95215a-f03a-4324-974e-1130203bf6f4.png)

**Reply by jitinnair1** [reply]
So sorry about this. I should have added clearer instructions!

This is happening because of the `url` field in the `_config.yml` file. In the `_config.yml` file, the `baseurl` will be "/" followed by the name of your repo.  (like `/repo_name`). The `url` field should be used in the case of custom domains only. So you should leave it only as `""` if you are using the `username.github.io` domain. (Something like [this](https://github.com/jitinnair1/gradfolio/blob/628a2e7986a0712dcae1683432e1a1f17779f020/_config.yml))

Also, when you change the `baseurl` to `/repo_name`, when you run `jekyll serve --incremental --trace`, the localhost URL will be something like `http://127.0.0.1:4000/repo_name`

**Reply by jitinnair1** [reply]
> So sorry about this. I should have added clearer instructions!
> 
> This is happening because of the `url` field in the `_config.yml` file. In the `_config.yml` file, the `baseurl` will be "/" followed by the name of your repo. (like `/repo_name`). The `url` field should be used in the case of custom domains only. So you should leave it only as `""` if you are using the `username.github.io` domain. (Something like [this](https://github.com/jitinnair1/gradfolio/blob/628a2e7986a0712dcae1683432e1a1f17779f020/_config.yml))
> 
> Also, when you change the `baseurl` to `/repo_name`, when you run `jekyll serve --incremental --trace`, the localhost URL will be something like `http://127.0.0.1:4000/repo_name`

This was in response to https://github.com/jitinnair1/gradfolio/discussions/12#discussioncomment-1547880

**Reply by jitinnair1** [reply]
> They seo tag files are conflicting but I don't know how to edit this without a gemfile. It's not in `config.yml`
> 
> ![conflctingSEO](https://user-images.githubusercontent.com/33698820/139123424-9f95215a-f03a-4324-974e-1130203bf6f4.png)

Please remove this plugin from the `_config.yml` file for now as a temporary fix. The plugin is probably not compatible with jekyll 4.x. I'll create an issue for now and check how to fix this.

**Reply by TheAmazinRaven** [reply]
> > So sorry about this. I should have added clearer instructions!
> > This is happening because of the `url` field in the `_config.yml` file. In the `_config.yml` file, the `baseurl` will be "/" followed by the name of your repo. (like `/repo_name`). The `url` field should be used in the case of custom domains only. So you should leave it only as `""` if you are using the `username.github.io` domain. (Something like [this](https://github.com/jitinnair1/gradfolio/blob/628a2e7986a0712dcae1683432e1a1f17779f020/_config.yml))
> > Also, when you change the `baseurl` to `/repo_name`, when you run `jekyll serve --incremental --trace`, the localhost URL will be something like `http://127.0.0.1:4000/repo_name`
> 
> This was in response to [#12 (reply in thread)](https://github.com/jitinnair1/gradfolio/discussions/12#discussioncomment-1547880)

Thank you once again for helping me with this, I really appreciate it. It is now saying liquid  syntax error on line 14 on this file, despite nothing being there. I got the same alert when doing `jekyll serve --incremental --trace` before and after commenting out the seo tag on that page.
![liguidSyntaxError](https://user-images.githubusercontent.com/33698820/139131655-5de55a40-90e2-4ba6-a84b-97145604cefa.png)

**Reply by jitinnair1** [reply]
Hey @TheAmazinRaven,

I tried reproducing this error, but after installing the missing plugins, `jekyll serve` runs for me without any errors. Looking at the error messages in the screenshots you shared: your version of ruby is 3.0.0. Also, the latest version of `jekyll-seo-tag` is 2.7.1 but even after you installed the plugin, the one that it is complaining about is 2.1.0. My hunch is that this is because of some incompatibility of the versions of ruby, gem and/or jekyll in use. Maybe updating everything will help. 

Sorry for not being of much help here. I tried looking up what may be causing this, but could not find anything that might help. In a future version of the theme, I plan to add a gem file. Maybe that will help with the compatibility issues. Also, if changing the `url` field helped (as mentioned [here](https://github.com/jitinnair1/gradfolio/discussions/12?sort=old#discussioncomment-1548024)) maybe you can edit directly on GitHub and see if it is rendered correctly on GitHub Pages

**Reply by TheAmazinRaven** [reply]
I'm sorry for the delayed reply. Thank you so much for these tips. Could I still come here for help once I've tried it all?

**Reply by jitinnair1** [reply]
Sure, no problem
